assayist/processor/base.py

```diff
@@ -165,6 +165,17 @@ def create_or_update_rpm_artifact_from_rpm_info(self, rpm_info):
         :return: an Artifact object
         :rtype: assayist.common.models.content.Artifact
         """
+        # If payloadhash isn't present, then the RPM info is incomplete and probably comes from
+        # some API call like getBuildrootListing
+        if 'payloadhash' not in rpm_info:
+            if 'rpm_id' in rpm_info:
+                rpm_id = rpm_info['rpm_id']
+            elif 'id' in rpm_info:
+                rpm_id = rpm_info['id']
+            else:
+                raise RuntimeError('The rpm_info did not contain an ID')
+            rpm_info = self.koji_session.getRPM(rpm_id)
```

Comment:
I was testing the main analyzer with this change and it results in a considerable slow-down because it has to fire off loads of extra requests to get the RPM information. Since we know we will need this info anyway, can we get it during the metadata download phase using a multicall? Something like this:

```diff
diff --git a/assayist/processor/utils.py b/assayist/processor/utils.py
index 90aa338..4c7e6e3 100644
--- a/assayist/processor/utils.py
+++ b/assayist/processor/utils.py
@@ -100,7 +100,15 @@ def download_build_data(build_identifier, output_dir='/metadata'):
 
     buildroot_components = {}
     for bid in sorted(buildroot_ids):
-        buildroot_components[bid] = koji.getBuildrootListing(bid)
+        buildroot_components[bid] = [a['rpm_id'] for a in koji.getBuildrootListing(bid)]
+
+    for bid, rpm_ids in buildroot_components.items():
+        koji.multicall = True
+        for rpm_id in rpm_ids:
+            koji.getRPM(rpm_id)
+
+        rpm_infos = koji.multiCall()
+        buildroot_components[bid] = [rpm_info[0] for rpm_info in rpm_infos]
 
     if buildroot_components:
         write_file(buildroot_components, output_dir, Analyzer.BUILDROOT_FILE)
```

Reply:
@mprpic I agree. Could we make this a separate PR?